### PR TITLE
finished advent of code day 2

### PR DESCRIPTION
**Pull request by nxnode**
I realized I was doubling complexity by creating a list with the for line by line and then immediately iterating over that list instead of just doing it all within the single iteration over line by line.

I KNOW the dictionary is actually not being used but I still want it since it seems like we are going to build off of it. But I simplified it to just the elf ID and total and removed the list of individual calories for each individual elf.

these are sentences that no one could ever imagine would be typed today by me to you!

### CODE REVIEW

**Comment by nxnode** [comment]
also, would it be wrong to merge a text file used as a data set to main. As opposed to branching from main and adding the same text file to each branch?

File: code_challenges/advent_of_code/day_1_part_2_top_3_elves.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# https://adventofcode.com/2022/day/1#part2
+
+
+def count_calories(path):
+    elf_calories = {1: {"total": 0}}
+    calories_sorted = {1: 0, 2: 0, 3: 0}
```
Comment (W-Ely) [inline code_challenges/advent_of_code/day_1_part_2_top_3_elves.py]: `calories_sorted = {1: 0, 2: 0, 3: 0}` is unused.

File: code_challenges/advent_of_code/day_1_part_2_top_3_elves.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    elf_id = 1
+    elf_calories["highest"] = 0
```
Comment (W-Ely) [inline code_challenges/advent_of_code/day_1_part_2_top_3_elves.py]: `elf_calories["highest"]` is unused

File: code_challenges/advent_of_code/day_1_part_2_top_3_elves.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    with open(path, "r", encoding="utf-8") as file:
+        lines = []
```
Comment (W-Ely) [inline code_challenges/advent_of_code/day_1_part_2_top_3_elves.py]: `lines = []` is appended to but never used.

**Review by W-Ely — CHANGES_REQUESTED** [review]
Great! Just remove the unused bits,  and add a readme entry.

**Review by W-Ely — APPROVED** [review]

**Pull request merged**